src/actions/controller.ts

```diff
@@ -1,47 +1,61 @@
-import * as vscode from "vscode";
+import { extensions } from "vscode";
 import { getLogger } from "../logger/logger";
 import { IAction } from "./interfaces";
 import { _performAction } from "./performer";
 import { getParameter } from '../apis/parameters';
-import { forEach, uniq, get, split, compact } from "lodash";
+import { ActionsFactory } from './actionsFactory';
+import * as _ from "lodash";
+import * as actionsConfig from './actionsConfig';
+
 
 export class ActionsController {
   private static readonly actions: IAction[] = [];
 
-  public static loadActions() {
-    vscode.extensions.all.forEach(extension => {
-      const extActions = get(extension, "packageJSON.BASContributes.actions", []);
-      extActions.forEach((action: IAction) => {
-        ActionsController.actions.push(action);
+  public static loadContributedActions() {
+    const logger = getLogger().getChildLogger({ label: "loadContributedActions" });
+    extensions.all.forEach(extension => {
+      const extensionActions = _.get(extension, "packageJSON.BASContributes.actions", []);
+      _.forEach(extensionActions, actionAsJson => {
+        try {
+          const action: IAction = ActionsFactory.createAction(actionAsJson, true);
+          ActionsController.actions.push(action);
+        } catch (error) {
+          logger.error(`Failed to create action ${JSON.stringify(actionAsJson)}: ${error}`);
+        }
       });
     });
   }
 
-  public static getAction(id: string) {
+  public static getAction(id: string): IAction | undefined {
     return ActionsController.actions.find(action => action.id === id);
   }
 
   public static async performActionsFromParams() {
-    const logger = getLogger().getChildLogger({label: "actionController"});
+    const logger = getLogger().getChildLogger({ label: "performActionsFromParams" });
     const actionsParam = await getParameter("actions");
-    logger.trace(`configuration - actions= ${actionsParam}`);
-    const actionsIds = uniq(compact(split(actionsParam, ",")));
-    actionsIds.forEach(actionId => {
+    const actionsIds = _.uniq(_.compact(_.split(actionsParam, ",")));
+    logger.trace(`configuration - actionsIds= ${actionsIds}`);
+      _.forEach(actionsIds, async actionId => {
       const action = ActionsController.getAction(actionId);
       if (action) {
-        void _performAction(action);
+        await _performAction(action);
       } else {
         logger.trace(`action ${actionId} not found`);
       }
     });
   }
 
   public static performScheduledActions() {
-    const actionsSettings = vscode.workspace.getConfiguration();
-    const actionsList: any[] = actionsSettings.get("actions", []);
-    forEach(actionsList, action => {
-      void _performAction(action);
+    const logger = getLogger().getChildLogger({ label: "performScheduledActions" });
```

Comment:
why do we need a childLogger for this one function?
Could we not just add an argument on the logging invocation?
The logger should be able to accept an **object** as the arguments.